### PR TITLE
5.0.0.12

### PR DESCRIPTION
**Pull request by Ghost-chu**

### CODE REVIEW

**Review by sandtechnology — CHANGES_REQUESTED** [review]

File: src/main/java/org/maxgamer/quickshop/util/matcher/item/BukkitItemMatcherImpl.java [hunk 1]
```diff
@@ -78,13 +81,13 @@ public boolean matches(@Nullable ItemStack original, @Nullable ItemStack tester)
...
-//                return true;
-//            }
-//        }
+        if (plugin.getNbtapi() != null) {
+            NBTItem nbtItemOriginal = new NBTItem(original);
+            NBTItem nbtItemTester = new NBTItem(tester);
+            if (Objects.equals(nbtItemOriginal.getString("shopId"), nbtItemTester.getString("shopId"))) {
+                return true;
+            }
+        }
```
[the remark this answers is not in the document]
Comment (Ghost-chu) [reply src/main/java/org/maxgamer/quickshop/util/matcher/item/BukkitItemMatcherImpl.java:90]: /minecraft:give @s minecraft:stone{shopItemId:"test"} to create a stone with test shopItemId
/minecraft:give @s minecraft:dirt{shopItemId:"test"} to create a dirt with test shopItemId

[the remark this answers is not in the document]
Comment (Ghost-chu) [reply src/main/java/org/maxgamer/quickshop/util/matcher/item/BukkitItemMatcherImpl.java:90]: /minecraft:give @s minecraft:stone{shopItemId:"test"} to create a stone with test shopItemId
/minecraft:give @s minecraft:dirt{shopItemId:"test"} to create a dirt with test shopItemId

File: src/main/java/org/maxgamer/quickshop/integration/advancedregionmarket/AdvancedShopRegionMarketIntegration.java [hunk 2]
```diff
@@ -136,9 +131,9 @@ public void unload() {
 
     private void handleDeletion(Region region) {
         Vector minPoint = region.getRegion().getMinPoint();
-        Vector maxPoint = region.getRegion().getMaxPoint();
+        Vector maxPoint = region.getRegion().getMinPoint();
         World world = region.getRegionworld();
-        Set<Chunk> chuckLocations = new HashSet<>();
+        Set<Chunk> chuckLocations = new HashSet<Chunk>();
```
Comment (sandtechnology) [inline src/main/java/org/maxgamer/quickshop/integration/advancedregionmarket/AdvancedShopRegionMarketIntegration.java:136]: Vector maxPoint = region.getRegion().getMinPoint() is wrong

File: src/main/java/org/maxgamer/quickshop/integration/advancedregionmarket/AdvancedShopRegionMarketIntegration.java [hunk 3]
```diff
@@ -155,12 +150,10 @@ private void handleDeletion(Region region) {
                 shopMap.putAll(shopsInChunk);
             }
         }
-        for (Map.Entry<Location, Shop> shopEntry : shopMap.entrySet()) {
-            Location shopLocation = shopEntry.getKey();
+        for (Location shopLocation : shopMap.keySet()) {
             if (region.getRegion().contains(shopLocation.getBlockX(), shopLocation.getBlockY(), shopLocation.getBlockZ())) {
-                Shop shop = shopEntry.getValue();
+                Shop shop = shopMap.get(shopLocation);
```
Comment (sandtechnology) [inline src/main/java/org/maxgamer/quickshop/integration/advancedregionmarket/AdvancedShopRegionMarketIntegration.java:155]: Still wrong revert commit

**Pull request merged**